### PR TITLE
Added articles on var, let and const, Javascript arrays and referenced the same in Javascript Roadmap. 

### PR DESCRIPTION
**Pull request by Sachin-12**
Added Javascript let,var,const and Arrays markdown files. Referenced the same in Javascript roadmap

### CODE REVIEW

File: docs/references/difference-between-var-let-and-const.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+```
+
+## const
+
+Keyword const is used to hold a `constant value` that can never be changed. Similar to let, const also has a `block-scope`.
+
+const cannot be redeclared or reassigned to a new value. This is beacuse const creates a read-only reference to a value. It doesn't mean that the value is immutable. `Primitive values` (string, number, boolean, null, undefined) are `immutable` while the `non-primitive values` (object, array, function) are `mutable`.
+
+Non
```
Comment (Sachin-12) [inline docs/references/difference-between-var-let-and-const.md]: Remove line 64

**Comment by DaniAkash** [comment]
@Sachin-12 this article goes in the Javascript roadmap not in the general roadmap

**Comment by DaniAkash** [comment]
@Sachin-12 don't close this pull request. Make changes in your fork and just push the code. This PR will be automatically updated.

File: docs/general-roadmap.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ The following topics are necessary regardless of whatever language you choose fo
     - YAGNI
   - Understanding different licenses
   - [Semantic Versioning](./references/semantic-versioning.md)
+  - [Differences Between Var, Let and Const](./references/difference-between-var-let-and-const.md)
```
Comment (DaniAkash) [inline docs/general-roadmap.md]: Just move this line into the Javascript roadmap

File: docs/references/difference-between-var-let-and-const.md (hunk 1 above)
Comment (DaniAkash) [reply docs/references/difference-between-var-let-and-const.md]: you can remove it and push the code

File: docs/general-roadmap.md (hunk 2 above)
Comment (Sachin-12) [reply docs/general-roadmap.md]: Sure dani.

Comment (Sachin-12) [reply docs/general-roadmap.md]: Sure @DaniAkash

**Comment by Sachin-12** [comment]
@DaniAkash  I have updated the article as per your suggestion.

**Pull request merged**